build_tools/github_actions/runner/config/setup.sh

```diff
@@ -14,15 +14,15 @@ set -euo pipefail
 
 echo "Register the self-hoster runner."
 chmod +x /home/runner/config/register.sh
-runuser --user runner ~runner/config/register.sh
+runuser --user runner /home/runner/config/register.sh
 
 echo "Setup the deregister service."
-cp ~runner/config/github-actions-runner-deregister.service /etc/systemd/system/
-chmod +x ~runner/config/deregister.sh
+cp /home/runner/config/github-actions-runner-deregister.service /etc/systemd/system/
+chmod +x /home/runner/config/deregister.sh
 
 echo "Setup the start actions runner service."
-cp ~runner/config/github-actions-runner-start.service /etc/systemd/system/
-chmod +x ~runner/config/start.sh
+cp /home/runner/config/github-actions-runner-start.service /etc/systemd/system/
+chmod +x /home/runner/config/start.sh
```

Comment:
`~foo` should be expanded to the home directory of user `foo`, regardless of who the current user is. Did you find that it didn't work?

Reply:
There was an issue encountered mid setup.sh. It seemed resolved when these path changes were tested locally, but I see how this works now and also appears to work just fine. Restored.